### ISSUE TITLE
Cellranger FastQ file name collision 

### ISSUE THREAD

**Issue by berguner** [nf-core/scrnaseq#147]
### Description of the bug

Hi,

The problem:
I tried running the pipeline (v2.0.0) for Cellranger count results on AWS. We sequenced the same samples on 2 sequencing runs for increased depth. The pipeline threw a file name collision error because the the Fastq file names are same even though the files have different S3 parent folders.

Solution recommendation:
`cellranger count` can deal with such situations by having the FastQ files in separate run folders. `cellranger` scans the upper folder to locate the Fastq files. Keeping the Fastq files in separate run folders might be a quick solution for this.

### Command used and terminal output

```console
Command:

nextflow run nf-core/scrnaseq \
        -r 2.0.0 \
        -c nextflow.config \
        --outdir s3://some-bucket \
        --input s3://some-bucket/fastq_list.nf-core_scrnaseq.csv \
        --aligner cellranger \
        --cellranger_index  s3://ref-bucket/ref/refdata-gex-GRCh38-2020-A/
```

Error message:
```
DEBUG nextflow.Session - Session aborted -- Cause: Process `NFCORE_SCRNASEQ:SCRNASEQ:CELLRANGER_ALIGN:CELLRANGER_COUNT` input file name collision -- There are multiple input files for each of the following file names: scRNA1927_10_S10_L001_R2_001.fastq.gz, scRNA1927_10_S10_L003_R2_001.fastq.gz, scRNA1927_10_S10_L003_R1_001.fastq.gz, scRNA1927_10_S10_L002_R2_001.fastq.gz, scRNA1927_10_S10_L002_R1_001.fastq.gz, scRNA1927_10_S10_L004_R1_001.fastq.gz, scRNA1927_10_S10_L004_R2_001.fastq.gz, scRNA1927_10_S10_L001_R1_001.fastq.gz
```
```


### Relevant files

_No response_

### System information

_No response_

**Comment by grst** [comment]
Hi @berguner, 

thanks for the report! Could you please share what your samplesheet looks like for the offending sample? 
I haven't worked with nextflow on aws so far and this would help me understand how the "folder structure" looks like when using S3 buckets.

**Comment by berguner** [comment]
Hi @grst ,

Below are the lines for one of the samples. 
```
sample,fastq_1,fastq_2
scRNA1927_1,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L001_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L001_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L002_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L002_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L003_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L003_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L004_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1985/220920_A01738_0031_AH325JDSX5/scRNA1927_1_S1_L004_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L001_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L001_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L002_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L002_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L003_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L003_R2_001.fastq.gz
scRNA1927_1,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L004_R1_001.fastq.gz,s3://my-fastq-bucket/SEQ1986/220920_A01738_0030_BH32CNDSX5/scRNA1927_1_S1_L004_R2_001.fastq.gz
```

Here we have additional parent folders "SEQXXXX" which are specific to our setup. Cellranger count scans one level of folders such as the illumina run folders (`220920_A01738_0030_BH32CNDSX5`, `220920_A01738_0031_AH325JDSX5`).

**Comment by berguner** [comment]
I tried running `cellranger count` locally and had to provide the list of FastQ parent folders explicitly with `--fastqs` parameter.

Here is the working command:
```
cellranger count \
        --id ${SAMPLE_NAME} \
        --fastqs "/data/working/fastq/SEQ1986,/data/working/fastq/SEQ1985" \
        --sample ${SAMPLE_NAME} \
        --transcriptome ${REF_DIR}
```

So, I had to give 1 upper folder (`SEQ1985`  , `SEQ1986`) of the FastQ folders (`220920_A01738_0030_BH32CNDSX5`, `220920_A01738_0031_AH325JDSX5`).

**Comment by grst** [comment]
I was wondering if we should instead merge the files manually (i.e. concatenate) before we pass them to cellranger, similar to what we do in the RNA-seq pipeline. I don't know if that would have any side effects, but I'd guess not. This was previously discussed here #92 and also somewhere on slack, but I can't find it atm. 

If this is not an option, we'd probably need an additional column in the samplesheet (e.g. `technical_replicate`) that can be used as the folder name during staging.

**Comment by berguner** [comment]
Concatenating the FastQ files would probably work without any side effects. One downside might be that this will create an additional task where the Fastq files are copied back and forth to the cloud storage for concatenation. It would be more efficient for the cloud to run concatenation in the same (`CELLRANGER_COUNT`) task.

My guess is that adding another column would be the most straightforward solution because this would let cellranger handle concatenation.

**Comment by grst** [comment]
Should finally be fixed on `dev` via #246.
